src/server/chromium/crNetworkManager.ts

```diff
@@ -442,7 +443,33 @@ class InterceptableRequest implements network.RouteDelegate {
     if (postDataEntries && postDataEntries.length && postDataEntries[0].bytes)
       postDataBuffer = Buffer.from(postDataEntries[0].bytes, 'base64');
 
-    this.request = new network.Request(allowInterception ? this : null, frame, redirectedFrom, documentId, url, type, method, postDataBuffer, headersObjectToArray(headers));
+    this.request = new network.Request(null, frame, redirectedFrom?.request || null, documentId, url, type, method, postDataBuffer, headersObjectToArray(headers));
+  }
+
+  _routeForRedirectChain(): RouteImpl | null {
+    let request: InterceptableRequest = this;
+    while (request._redirectedFrom)
+      request = request._redirectedFrom;
+    return request._route;
+  }
+}
+
+class RouteImpl implements network.RouteDelegate {
+  private readonly _client: CRSession;
+  private _interceptionId: string;
+  private _responseInterceptedPromise: Promise<{ event: Protocol.Fetch.requestPausedPayload; redirectedRequest: InterceptableRequest; }>;
+  private _responseInterceptedCallback: ((data: { event: Protocol.Fetch.requestPausedPayload, redirectedRequest: InterceptableRequest }) => void) = () => {};
+  _interceptingResponse: boolean = false;
+
+  constructor(client: CRSession, interceptionId: string) {
+    this._client = client;
+    this._interceptionId = interceptionId;
+    this._responseInterceptedPromise = new Promise(resolve => this._responseInterceptedCallback = resolve);
+  }
+
+  _onInterceptedResponse(data: {event: Protocol.Fetch.requestPausedPayload, redirectedRequest: InterceptableRequest}) {
+    this._interceptionId = data.redirectedRequest._interceptionId!;
+    this._responseInterceptedCallback(data);
```

Comment:
I still think we should merge _responseInterceptedCallback and _onIntercepted. At least, pass two parameters to `onIntercepted` instead of `data`.

Reply:
Done.